### PR TITLE
Tycho 1.0

### LINKED ISSUE #1598: Update to Tycho 1.0
Toolchains aren't doing what we need: although our bundles declare `BREE: JavaSE-1.7`, some of the classes we use, like `org.eclipse.core.databinding.observable.value.ComputedValue`, reference and use Java 8 classes in Neon and Oxygen.  So compilation will fail:
```
$ mvn -Declipse.target=neon -Dmaven.repo.local=/tmp/m2 package
[…]
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:0.24.0:compile (default-compile) on project com.google.cloud.tools.eclipse.appengine.deploy.ui: Compilation failure: Compilation failure:
[ERROR] /Users/bsd/Manumitting/Projects/GooglePluginEclipse/ide-target-platform/plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/StandardDeployPreferencesPanel.java:[1]
[ERROR] /Users/bsd/Manumitting/Projects/GooglePluginEclipse/ide-target-platform/plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/StandardDeployPreferencesPanel.java:[1]
[ERROR] /*
[ERROR] ^
[ERROR] The type java.util.function.Supplier cannot be resolved. It is indirectly referenced from required .class files
[ERROR] -> [Help 1]
```
Since we ensure Travis builds with 1.7 against Mars, we should just abandon the use of toolchains.

### PR DESCRIPTION
**Pull request by briandealwis**
Update build to use Tycho 1.0.0, Maven 3.5.0 (not actually required), and Cloud SDK 150.0.0.

Tycho 1.0 was recently been released and requires Java 8 [as of 0.25](https://bugs.eclipse.org/bugs/show_bug.cgi?id=485427).  This causes some complications as we previously relied on building with Java 7 on Travis to catch any Java-8'isms.  And `tycho-compiler-plugin`'s support for using the bundle's BREE doesn't work for Neon or later as they require a Java 8 and we hit class resolution problems when compiling with a Java 7 JDK.  So this change does two things:
  1. It configures the the `tycho-compiler-plugin` to only use the BREE compilation when building with a JavaSE-1.7.
  2. It leverages `tycho-surefire-plugin`'s [default configuration of `useJDK=SYSTEM`](https://eclipse.org/tycho/sitedocs/tycho-surefire/tycho-surefire-plugin/test-mojo.html#useJDK) to run the tests using the configured Maven toolchain, which we configure to use a Java 7 runtime for Mars and Java 8 for Neon and Oxygen; I've included an example log below to show how a Java 8 project fails. 
This should catch any accidental Java 8'isms and Java 9'isms!

Tycho 1.0 caught a missing export-package in google apis, and some missing imports and an explicit Tycho version reference in `dataflow`. So I also took the opportunity to remove unnecessary dataflow configuration information that is obtained from the parent.

This change also does the following:
    - uses Maven's `-B` batch-mode to cut down on log messages
    - changes the default `tycho.toolchains` value to SYSTEM, rendering #685 obsolete, since Travis will pick up any issues
    - bumps Google Cloud SDK to 150.0.0

Fixes #1598

Log showing how converting `com.google.cloud.tools.eclipse.appengine.compat` to Java 8 causes a test failure (some lines elided):
```
[INFO] --- tycho-surefire-plugin:1.0.0:test (default-test) @ com.google.cloud.tools.eclipse.appengine.compat.test ---
[INFO] Toolchain in tycho-surefire-plugin: JDK[/usr/lib/jvm/java-7-oracle/jre]
[INFO] Expected eclipse log file: /home/travis/build/GoogleCloudPlatform/google-cloud-eclipse/plugins/com.google.cloud.tools.eclipse.appengine.compat.test/target/work/data/.metadata/.log
[INFO] Command line:
	[/usr/lib/jvm/java-7-oracle/jre/bin/java, -Dosgi.noShutdown=false, -Dosgi.os=linux, -Dosgi.ws=gtk, -Dosgi.arch=x86_64, -Dorg.eclipse.swtbot.search.timeout=30000, -Xms40m, -Xmx1G, -XX:MaxPermSize=512m, -Djava.awt.headless=true, -Dosgi.clean=true, -jar, /home/travis/.m2/repository/p2/osgi/bundle/org.eclipse.equinox.launcher/1.3.100.v20150511-1540/org.eclipse.equinox.launcher-1.3.100.v20150511-1540.jar, -debug, -consolelog, -data, /home/travis/build/GoogleCloudPlatform/google-cloud-eclipse/plugins/com.google.cloud.tools.eclipse.appengine.compat.test/target/work/data, -install, /home/travis/build/GoogleCloudPlatform/google-cloud-eclipse/plugins/com.google.cloud.tools.eclipse.appengine.compat.test/target/work, -configuration, /home/travis/build/GoogleCloudPlatform/google-cloud-eclipse/plugins/com.google.cloud.tools.eclipse.appengine.compat.test/target/work/configuration, -application, org.eclipse.tycho.surefire.osgibooter.headlesstest, -testproperties, /home/travis/build/GoogleCloudPlatform/google-cloud-eclipse/plugins/com.google.cloud.tools.eclipse.appengine.compat.test/target/surefire.properties]
Configuration location:
    file:/home/travis/build/GoogleCloudPlatform/google-cloud-eclipse/plugins/com.google.cloud.tools.eclipse.appengine.compat.test/target/work/configuration/
Configuration file:
    file:/home/travis/build/GoogleCloudPlatform/google-cloud-eclipse/plugins/com.google.cloud.tools.eclipse.appengine.compat.test/target/work/configuration/config.ini loaded
Install location:
    file:/home/travis/build/GoogleCloudPlatform/google-cloud-eclipse/plugins/com.google.cloud.tools.eclipse.appengine.compat.test/target/work/
Framework located:
    file:/home/travis/.m2/repository/p2/osgi/bundle/org.eclipse.osgi/3.10.102.v20160118-1700/org.eclipse.osgi-3.10.102.v20160118-1700.jar
Framework classpath:
    file:/home/travis/.m2/repository/p2/osgi/bundle/org.eclipse.osgi/3.10.102.v20160118-1700/org.eclipse.osgi-3.10.102.v20160118-1700.jar
Debug options:
    file:/home/travis/build/GoogleCloudPlatform/google-cloud-eclipse/plugins/com.google.cloud.tools.eclipse.appengine.compat.test/.options not found
Time to load bundles: 2260
!SESSION 2017-04-11 19:49:16.527 -----------------------------------------------
eclipse.buildId=unknown
java.version=1.7.0_76
java.vendor=Oracle Corporation
BootLoader constants: OS=linux, ARCH=x86_64, WS=gtk, NL=en_US
Framework arguments:  -application org.eclipse.tycho.surefire.osgibooter.headlesstest -testproperties /home/travis/build/GoogleCloudPlatform/google-cloud-eclipse/plugins/com.google.cloud.tools.eclipse.appengine.compat.test/target/surefire.properties
Command-line arguments:  -debug -consolelog -data /home/travis/build/GoogleCloudPlatform/google-cloud-eclipse/plugins/com.google.cloud.tools.eclipse.appengine.compat.test/target/work/data -application org.eclipse.tycho.surefire.osgibooter.headlesstest -testproperties /home/travis/build/GoogleCloudPlatform/google-cloud-eclipse/plugins/com.google.cloud.tools.eclipse.appengine.compat.test/target/surefire.properties
!ENTRY com.google.cloud.tools.eclipse.appengine.compat 4 0 2017-04-11 19:49:19.899
!MESSAGE FrameworkEvent ERROR
!STACK 0
org.osgi.framework.BundleException: Could not resolve module: com.google.cloud.tools.eclipse.appengine.compat [6]
  Unresolved requirement: Require-Capability: osgi.ee; filter:="(&(osgi.ee=JavaSE)(version=1.8))"
	at org.eclipse.osgi.container.Module.start(Module.java:434)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.incStartLevel(ModuleContainer.java:1582)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.incStartLevel(ModuleContainer.java:1561)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.doContainerStartLevel(ModuleContainer.java:1533)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.dispatchEvent(ModuleContainer.java:1476)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.dispatchEvent(ModuleContainer.java:1)
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230)
	at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:340)
!ENTRY com.google.cloud.tools.eclipse.test.util 4 0 2017-04-11 19:49:19.901
!MESSAGE FrameworkEvent ERROR
!STACK 0
org.osgi.framework.BundleException: Could not resolve module: com.google.cloud.tools.eclipse.test.util [16]
  Unresolved requirement: Import-Package: com.google.cloud.tools.eclipse.appengine.facets
    -> Export-Package: com.google.cloud.tools.eclipse.appengine.facets; bundle-version="0.1.0.201704111618"; bundle-symbolic-name="com.google.cloud.tools.eclipse.appengine.facets"; version="0.0.0"
       com.google.cloud.tools.eclipse.appengine.facets [32]
         Unresolved requirement: Import-Package: com.google.cloud.tools.eclipse.appengine.compat
           -> Export-Package: com.google.cloud.tools.eclipse.appengine.compat; bundle-version="0.1.0.201704111940"; bundle-symbolic-name="com.google.cloud.tools.eclipse.appengine.compat"; version="0.0.0"
              com.google.cloud.tools.eclipse.appengine.compat [6]
                Unresolved requirement: Require-Capability: osgi.ee; filter:="(&(osgi.ee=JavaSE)(version=1.8))"
	at org.eclipse.osgi.container.Module.start(Module.java:434)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.incStartLevel(ModuleContainer.java:1582)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.incStartLevel(ModuleContainer.java:1561)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.doContainerStartLevel(ModuleContainer.java:1533)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.dispatchEvent(ModuleContainer.java:1476)
	at org.eclipse.osgi.container.ModuleContainer$ContainerStartLevel.dispatchEvent(ModuleContainer.java:1)
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230)
	at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:340)
Starting application: 5019
-------------------------------------------------------
 T E S T S
-------------------------------------------------------
!ENTRY org.eclipse.swt.win32.win32.x86_64 2 0 2017-04-11 19:49:21.581
!MESSAGE Could not resolve module: org.eclipse.swt.win32.win32.x86_64 [216]
  Unresolved requirement: Require-Capability: eclipse.platform; filter:="(& (osgi.ws=win32) (osgi.os=win32) (osgi.arch=x86_64))"
!ENTRY org.eclipse.swt.cocoa.macosx.x86_64 2 0 2017-04-11 19:49:21.582
!MESSAGE Could not resolve module: org.eclipse.swt.cocoa.macosx.x86_64 [217]
  Unresolved requirement: Require-Capability: eclipse.platform; filter:="(& (osgi.ws=cocoa) (osgi.os=macosx) (osgi.arch=x86_64) )"
!ENTRY org.eclipse.swt.gtk.linux.x86 2 0 2017-04-11 19:49:21.582
!MESSAGE Could not resolve module: org.eclipse.swt.gtk.linux.x86 [62]
  Unresolved requirement: Require-Capability: eclipse.platform; filter:="(& (osgi.ws=gtk) (osgi.os=linux) (osgi.arch=x86))"
!ENTRY com.google.cloud.tools.eclipse.appengine.compat.test 2 0 2017-04-11 19:49:21.583
!MESSAGE Could not resolve module: com.google.cloud.tools.eclipse.appengine.compat.test [14]
  Unresolved requirement: Import-Package: com.google.cloud.tools.eclipse.test.util.project
    -> Export-Package: com.google.cloud.tools.eclipse.test.util.project; bundle-version="0.1.0.201703291554"; bundle-symbolic-name="com.google.cloud.tools.eclipse.test.util"; version="0.0.0"
       com.google.cloud.tools.eclipse.test.util [16]
         Unresolved requirement: Import-Package: com.google.cloud.tools.eclipse.appengine.facets
           -> Export-Package: com.google.cloud.tools.eclipse.appengine.facets; bundle-version="0.1.0.201704111618"; bundle-symbolic-name="com.google.cloud.tools.eclipse.appengine.facets"; version="0.0.0"
              com.google.cloud.tools.eclipse.appengine.facets [32]
                Unresolved requirement: Import-Package: com.google.cloud.tools.eclipse.appengine.compat
                  -> Export-Package: com.google.cloud.tools.eclipse.appengine.compat; bundle-version="0.1.0.201704111940"; bundle-symbolic-name="com.google.cloud.tools.eclipse.appengine.compat"; version="0.0.0"
                     com.google.cloud.tools.eclipse.appengine.compat [6]
                       Unresolved requirement: Require-Capability: osgi.ee; filter:="(&(osgi.ee=JavaSE)(version=1.8))"
!ENTRY com.google.cloud.tools.eclipse.appengine.compat 2 0 2017-04-11 19:49:21.583
!MESSAGE Could not resolve module: com.google.cloud.tools.eclipse.appengine.compat [6]
  Unresolved requirement: Require-Capability: osgi.ee; filter:="(&(osgi.ee=JavaSE)(version=1.8))"
!ENTRY org.eclipse.swt.win32.win32.x86 2 0 2017-04-11 19:49:21.584
!MESSAGE Could not resolve module: org.eclipse.swt.win32.win32.x86 [215]
  Unresolved requirement: Require-Capability: eclipse.platform; filter:="(& (osgi.ws=win32) (osgi.os=win32) (osgi.arch=x86))"
!ENTRY com.google.cloud.tools.eclipse.test.util 2 0 2017-04-11 19:49:21.584
!MESSAGE Could not resolve module: com.google.cloud.tools.eclipse.test.util [16]
  Unresolved requirement: Import-Package: com.google.cloud.tools.eclipse.appengine.facets
    -> Export-Package: com.google.cloud.tools.eclipse.appengine.facets; bundle-version="0.1.0.201704111618"; bundle-symbolic-name="com.google.cloud.tools.eclipse.appengine.facets"; version="0.0.0"
       com.google.cloud.tools.eclipse.appengine.facets [32]
         Unresolved requirement: Import-Package: com.google.cloud.tools.eclipse.appengine.compat
           -> Export-Package: com.google.cloud.tools.eclipse.appengine.compat; bundle-version="0.1.0.201704111940"; bundle-symbolic-name="com.google.cloud.tools.eclipse.appengine.compat"; version="0.0.0"
              com.google.cloud.tools.eclipse.appengine.compat [6]
                Unresolved requirement: Require-Capability: osgi.ee; filter:="(&(osgi.ee=JavaSE)(version=1.8))"
!ENTRY com.google.cloud.tools.eclipse.appengine.facets 2 0 2017-04-11 19:49:21.585
!MESSAGE Could not resolve module: com.google.cloud.tools.eclipse.appengine.facets [32]
  Unresolved requirement: Import-Package: com.google.cloud.tools.eclipse.appengine.compat
    -> Export-Package: com.google.cloud.tools.eclipse.appengine.compat; bundle-version="0.1.0.201704111940"; bundle-symbolic-name="com.google.cloud.tools.eclipse.appengine.compat"; version="0.0.0"
       com.google.cloud.tools.eclipse.appengine.compat [6]
         Unresolved requirement: Require-Capability: osgi.ee; filter:="(&(osgi.ee=JavaSE)(version=1.8))"
!ENTRY org.eclipse.osgi 4 0 2017-04-11 19:49:21.585
!MESSAGE Application error
!STACK 1
java.lang.RuntimeException: Bundle com.google.cloud.tools.eclipse.appengine.compat.test is not found
	at org.eclipse.tycho.surefire.osgibooter.OsgiSurefireBooter.getBundleClassLoader(OsgiSurefireBooter.java:166)
	at org.eclipse.tycho.surefire.osgibooter.OsgiSurefireBooter.createCombinedClassLoader(OsgiSurefireBooter.java:102)
	at org.eclipse.tycho.surefire.osgibooter.OsgiSurefireBooter.run(OsgiSurefireBooter.java:95)
	at org.eclipse.tycho.surefire.osgibooter.HeadlessTestApplication.run(HeadlessTestApplication.java:21)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.eclipse.equinox.internal.app.EclipseAppContainer.callMethodWithException(EclipseAppContainer.java:587)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:198)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:380)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:235)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:669)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:608)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1515)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1488)
An error has occurred. See the log file
/home/travis/build/GoogleCloudPlatform/google-cloud-eclipse/plugins/com.google.cloud.tools.eclipse.appengine.compat.test/target/work/configuration/1491940156755.log.
[…]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 10:14 min
[INFO] Finished at: 2017-04-11T19:56:38Z
[INFO] Final Memory: 169M/501M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-surefire-plugin:1.0.0:test (default-test) on project com.google.cloud.tools.eclipse.appengine.compat.test: An unexpected error occured while launching the test runtime (return code 13). See log for details. -> [Help 1]
[…]
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,12 +59,18 @@ $ mvn -Declipse.target=neon package
 
 ### Configuring Maven/Tycho Toolchains
 
+_Note: this section is only relevant for configuring CI builds_
```
Comment (akerekes) [inline README.md]: I'd add this to the section title like `Configuring Maven/Tycho Toolchains in CI builds`

File: plugins/com.google.cloud.tools.eclipse.dataflow.core/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -8,12 +8,15 @@ Bundle-Vendor: %providerName
 Bundle-ClassPath: .
 Bundle-RequiredExecutionEnvironment: JavaSE-1.7
 Bundle-ActivationPolicy: lazy
-Export-Package: com.google.cloud.tools.eclipse.dataflow.core.launcher,
+Export-Package: com.google.cloud.tools.eclipse.dataflow.core,
+ com.google.cloud.tools.eclipse.dataflow.core.internal.properties,
```
Comment (akerekes) [inline plugins/com.google.cloud.tools.eclipse.dataflow.core/META-INF/MANIFEST.MF]: could this also be restricted to `x-friends`?

File: pom.xml [hunk 3]
```diff
@@ -502,12 +497,65 @@
...
+      using Maven and Tycho's support for toolchains.
+    -->
+    <profile>
+       <!-- 
+          We can only use tycho-compiler-plugin's `useJDK=BREE` mode
+          (which builds a bundle against a JDK that matches its
+          Bundle-RequiredExecutionEnvironment) when we're building
+          against Mars as Eclipse 4.6/Neon requires Java 8.
+      -->
+      <id>build-with-java-runtime-</id>
```
Comment (akerekes) [inline pom.xml]: `build-with-java-runtime-7`?

File: plugins/com.google.cloud.tools.eclipse.dataflow.core/META-INF/MANIFEST.MF (hunk 2 above)
Comment (briandealwis) [inline plugins/com.google.cloud.tools.eclipse.dataflow.core/META-INF/MANIFEST.MF]: I was overzealous: `.internal.properties isn't` necessary to export.

File: pom.xml (hunk 3 above)
Comment (briandealwis) [inline pom.xml]: I renamed it to `building-with-minimal-java-runtime` and adjusted the comments.

**Pull request merged**